sentry_streams/src/gcs_writer.rs

```diff
@@ -0,0 +1,66 @@
+use anyhow::anyhow;
+use pyo3::types::PyAnyMethods;
+use pyo3::types::PyBytes;
+use pyo3::Python;
+use reqwest::header::{AUTHORIZATION, CONTENT_TYPE};
+use reqwest::Client;
+use sentry_arroyo::processing::strategies::run_task_in_threads::{
+    RunTaskError, RunTaskFunc, TaskRunner,
+};
+
+use crate::routes::RoutedValue;
+use pyo3::prelude::*;
+use sentry_arroyo::types::Message;
+pub struct GCSWriter {}
+
+impl GCSWriter {
+    pub fn new() -> Self {
+        GCSWriter {}
+    }
+}
+
+fn to_bytes(payload: &RoutedValue) -> Vec<u8> {
+    let payload = Python::with_gil(|py| {
+        let payload = payload.payload.clone_ref(py);
+        // if payload.is_none(py) {
+        // prolly need to handle this case
+
+        let py_bytes: &Bound<PyBytes> = payload.bind(py).downcast().unwrap();
+        py_bytes.as_bytes().to_vec()
+    });
+    payload
+}
+
+// start by supporting bytes only
+impl TaskRunner<RoutedValue, RoutedValue, anyhow::Error> for GCSWriter {
```

Comment:
I'd advise you to assume your data will be a `Vec<u8>`. We can serialize into Vec<u8> before this step so we can avoid taking the gil to turn PyAny into vec.